modules/plugin/geotiff/src/test/java/org/geotools/gce/geotiff/GeoTiffReaderTest.java

```diff
@@ -1309,6 +1309,36 @@ public void testScaleOffset() throws IllegalArgumentException, IOException, Fact
         }
     }
 
+    @Test
+    public void testNodataFloatCast() throws IOException {
+        // has a very large nodata value with different values in float and double
+        final File scaleOffset =
+                TestData.file(GeoTiffReaderTest.class, "float32_nodata_negmax.tif");
+        GeoTiffReader reader = new GeoTiffReader(scaleOffset);
+
+        // read with explicit request not to rescale
```

Comment:
I think this comments comes from a test used as starting point. Is it still needed?
Does it involves any scalingOffset? (I see the file is named scaleOffset).
Anyway, nothing against that... just asking.